packages/frontend/src/app/directives/toggleable-password.directive.ts

```diff
@@ -0,0 +1,37 @@
+import { Directive, ElementRef, Renderer2 } from "@angular/core";
+
+@Directive({
+  selector: "[appToggleablePassword]",
+})
+export class ToggleablePasswordDirective {
+  private showPassword = false;
+  private eyeIcon!: HTMLElement;
+
+  constructor(
+    private el: ElementRef,
+    private renderer: Renderer2,
+  ) {
+    this.setup();
+  }
+
+  setup() {
+    // Create the eye icon for showing/hiding password
+    this.eyeIcon = this.renderer.createElement("ion-icon");
```

Comment:
I noticed while testing that this needs cursor: pointer. For accessibility, it'd also be nice for this to be a button (which would solve the problem anyway). You could use the icon-only button [here](https://ionicframework.com/docs/api/button#icons)